go.mod

```diff
@@ -312,3 +314,5 @@ replace github.com/bradfitz/gomemcache => github.com/themihai/gomemcache v0.0.0-
 replace github.com/thanos-io/thanos v0.22.0 => github.com/thanos-io/thanos v0.19.1-0.20211126105533-c5505f5eaa7d
 
 replace github.com/cloudflare/cloudflare-go => github.com/cyriltovena/cloudflare-go v0.27.1-0.20211118103540-ff77400bcb93
+
+exclude k8s.io/client-go v8.0.0+incompatible
```

Comment:
I'm trying to understand this `exclude` directive -- can you help me understand what it helps with?

Reply:
Adding exclude `k8s.io/client-go v8.0.0+incompatible` helps to keep version of github.com/influxdata/telegraf v1.16.3.
When I upgrade k8s.io/client-go to v0.23.6, go downgrades github.com/influxdata/telegraf to v0.10.1. When I upgrade telegraf back to v1.16.3, go upgrades k8s.io/client-go to v8.0.0+incompatible. But version v8.0.0+incompatible is actually older than v0.23.6.

`exclude k8s.io/client-go v8.0.0+incompatible` added to have both github.com/influxdata/telegraf v1.16.3 and k8s.io/client-go v0.23.6. 
I took this solution from prometheus https://github.com/prometheus/prometheus/blob/main/go.mod#L102-L103.

Not sure if we need to exclude other pre-go-mod kubernetes tags here. I decided to go with as small diff as possible in this PR

Reply:
Instead of ignoring `k8s.io/client-go` to fix that, can we pin the telegraf version instead?

Reply:
Ok, I think that's reasonable :+1: